### ISSUE TITLE
Score() vs. Correlation coefficient

### ISSUE THREAD

**Issue by jwgda** [pjaselin/Cubist#72]
Is there an explanation for what the meaning of the value returned by the score() function vs. the 'Correlation coefficient' that is displayed in the 'Evaluation of training data' from the Cubist output when verbose mode is enabled?

**Comment by DavidWalz** [comment]
Thank you for building this package @pjaselin! 
Don't forget the R² (coefficient of determination) is not the same as the Pearson correlation coefficient r between observed and predicted values, and it's only for linear models that one is the square of the other.

**Comment by jwgda** [comment]
Thanks @DavidWalz -- so does that mean the score() method then in fact does return the Coefficient of determination (R^2)?

I tried another simple case (1 rule, 1 committee) with the following dataset (two independent variables)

```
X1	X2	Y
0.884	3.741	9.16
0.877	3.784	10.23
0.883	3.381	9.85
0.888	3.943	11.36
0.89	3.734	10.86
0.878	3.647	10.42
0.877	3.544	10.73
0.88	3.607	10.73
0.886	4	11.36
0.879	3.584	10.36
0.886	3.702	10.8
0.876	2.173	8.6
0.867	3.314	10.29
0.867	3.997	11.17
0.889	3.914	12.05
0.884	3.845	12.74
0.877	3.887	12.68
0.881	4.02	12.3
0.885	4.237	12.43
```
Which returned the following:

`Cubist [Release 2.07 GPL Edition]  Mon Nov  1 15:52:45 2021
---------------------------------

    Target attribute `outcome'

Read 19 cases (3 attributes) from undefined.data

Model:

  Rule 1: [19 cases, mean 10.954535, range 8.599112 to 12.74175, est err 0.642506]

	outcome = 3.710855 + 1.94 X2


Evaluation on training data (19 cases):

    Average  |error|           0.890451
    Relative |error|               1.00
    Correlation coefficient        0.33


	Attribute usage:
	  Conds  Model

	         100%    X2


Time: 0.0 secs


score():  0.9999999999999332`

Which seems like it's only using one variable -- but the score function returns an absurdly high number for R^2: 0.999.

I tried dropping the X1 variable, since it wasn't selected anyway, and got this:

`Cubist [Release 2.07 GPL Edition]  Mon Nov  1 15:53:07 2021
---------------------------------

    Target attribute `outcome'

Read 19 cases (2 attributes) from undefined.data

Model:

  Rule 1: [19 cases, mean 10.954535, range 8.599112 to 12.74175, est err 0.642506]

	outcome = 3.710855 + 1.94 X2


Evaluation on training data (19 cases):

    Average  |error|           0.548818
    Relative |error|               0.62
    Correlation coefficient        0.81


	Attribute usage:
	  Conds  Model

	         100%    X2


Time: 0.0 secs


score():  0.8244528284399067
`
It seems like it came up with the same rules, but with totally different (but more plausible) error values -- why is that?

With the first case (with the high R^2), I tried back-predicting the training values and got nearly 0 error -- something doesn't seem right there.

**Comment by DavidWalz** [comment]
> so does that mean the score() method then in fact does return the Coefficient of determination (R^2)?

yes, see https://scikit-learn.org/stable/modules/generated/sklearn.base.RegressorMixin.html
and yes, there seems to be something fishy here.

**Comment by pjaselin** [comment]
Hi @DavidWalz and @jwgda, I think I figured it out. In my next update, I exposed the feature from the C library to choose whether to use composite (instance + rule) models. It looks like the correlation coefficient is the Pearson correlation coefficient based on: https://github.com/pjaselin/Cubist/blob/44b14a1760c2e33e33f86a78011143ba46125b2c/cubist/src/construct.c#L322 and also by using: `pearsonr` from scipy with composite=False. When a composite model is used, then this agreement no longer holds true.
